back-end/src/main/java/box/BoxGetter.java

```diff
@@ -11,80 +11,40 @@
-        double oldMinx = oldBox.getMinx(), oldMiny = oldBox.getMiny();
-        double oldMaxx = oldBox.getMaxx(), oldMaxy = oldBox.getMaxy();
-
-        // calculate delta area
+        // WKT stuff for all layers
         GeometryFactory fact = new GeometryFactory();
         WKTReader wktRdr = new WKTReader(fact);
-        String wktNew =
-                "POLYGON(("
-                        + newMinx
-                        + " "
-                        + newMiny
-                        + ","
-                        + newMinx
-                        + " "
-                        + newMaxy
-                        + ","
-                        + newMaxx
-                        + " "
-                        + newMaxy
-                        + ","
-                        + newMaxx
-                        + " "
-                        + newMiny
-                        + ","
-                        + newMinx
-                        + " "
-                        + newMiny
-                        + "))";
-        String wktOld =
-                "POLYGON(("
-                        + oldMinx
-                        + " "
-                        + oldMiny
-                        + ","
-                        + oldMinx
-                        + " "
-                        + oldMaxy
-                        + ","
-                        + oldMaxx
-                        + " "
-                        + oldMaxy
-                        + ","
-                        + oldMaxx
-                        + " "
-                        + oldMiny
-                        + ","
-                        + oldMinx
-                        + " "
-                        + oldMiny
-                        + "))";
-        Geometry newBoxGeom = wktRdr.read(wktNew);
-        Geometry oldBoxGeom = wktRdr.read(wktOld);
-        Geometry deltaGeom = newBoxGeom.difference(oldBoxGeom);
         WKTWriter wktWtr = new WKTWriter();
-        String deltaWkt = wktWtr.write(deltaGeom);
+        String wktNew = newBox.getWKT();
+        Geometry newBoxGeom = wktRdr.read(wktNew);
 
         // loop through each layer
         for (int i = 0; i < c.getLayers().size(); i++) {
             Layer curLayer = c.getLayers().get(i);
             // if this layer is static, add an empty placeholder
-            if (curLayer.isStatic()) data.add(new ArrayList<>());
-            else
-                data.add(
-                        curLayer.getIndexer()
-                                .getDataFromRegion(
-                                        c, i, deltaWkt, predicates.get(i), newBox, oldBox));
+            if (curLayer.isStatic() || !curLayer.getFetchingScheme().equals("dbox")) {
+                data.add(new ArrayList<>());
+                continue;
+            }
+            // calculate delta area
+            Box curOldBox;
+            if (curLayer.isDeltaBox()) curOldBox = oldBox;
+            else curOldBox = DUMMY_OLD_BOX;
```

Comment:
is this dummy box only used upon initialization when there is no old box?

Reply:
Actually, no. 

This `DUMMY_OLD_BOX` is used in the backend for dynamic boxes with no delta box, in which case the backend will ignore the old box sent by the front-end. 

Note that previously the front-end will send an old box if the `deltaBox` is configure to be globally true. Now that it's configured per layer, I let the front-end always send an old box (will be dummy values when there is no old box), and let the back-end use it for layers that require `deltaBox`. So for layers that do not require `deltaBox`, the backend will use this `DUMMY_OLD_BOX` and ignore the old box sent by the front-end.